crates/trigger/src/stdio.rs

```diff
@@ -223,32 +283,57 @@ impl AsyncWrite for ComponentStdioWriter {
         cx: &mut std::task::Context<'_>,
     ) -> Poll<std::result::Result<(), std::io::Error>> {
         let this = self.get_mut();
-        match this.state {
-            ComponentStdioWriterState::File => {
-                std::pin::Pin::new(&mut this.async_file).poll_shutdown(cx)
-            }
-            ComponentStdioWriterState::Follow(_) => {
+
+        match &mut this.inner {
+            ComponentStdioWriterInner::Inherit => {
                 std::pin::Pin::new(&mut tokio::io::stderr()).poll_flush(cx)
             }
+            ComponentStdioWriterInner::Forward {
+                async_file, state, ..
+            } => match state {
+                ComponentStdioWriterState::File => std::pin::Pin::new(async_file).poll_shutdown(cx),
+                ComponentStdioWriterState::Follow(_) => {
```

Comment:
This is at least the second function where `Inherit` and `Forward { state = Follow }` have the same behaviour.  Should we have a helper function on `ComponentStdioWriterInner` that returns "use stderr or this handle"?  Or the destination stream so we can really unify the cases?

Reply:
I'm not really following your suggestions, could you please elaborate?

Reply:
Don't worry about it, it just seemed as if we might be in a situation where two of the three states had the same behaviour and I was just wondering if we could extract that shared behaviour.